src/pybind/mgr/volumes/fs/subvolume.py

```diff
@@ -77,6 +77,9 @@ def create_subvolume(self, spec, size=None, namespace_isolated=True, mode=0o755,
 
         try:
             if size is not None:
+                if size < 0:
```

Comment:
Instead can we catch the InvalidValue exception raised by setxattr, and raise a VolumeException with an appropriate message?

Reply:
> Does setxattr let you do that?

Yes `setxattr` was allowing `0`. But it is fixed now.

Reply:
@ajarr Pushed the fix. The testing shows that this code still allows setting size `0`, but prevents negative values.